src/search.c

```diff
@@ -4675,6 +4676,13 @@ fuzzy_match_in_list(
 	ptrs[i].idx = i;
 	ptrs[i].item = li;
 	ptrs[i].score = SCORE_NONE;
+
+	if (num_limit > 0 && found_match >= num_limit)
```

Comment:
One issue with placing a limit on the maximum number of matches is that the best match may be found after this limit. 
 In that case, matchfuzzy() will return the best match within this limit.